package.json

```diff
@@ -842,6 +842,13 @@
                     "group": "z_externalLinks@1"
                 }
             ],
+            "explorer/context": [
+                {
+                    "command": "aws.deploySamApplication",
+                    "when": "isFileSystemResource == true && resourceFilename =~ /^template\\.(json|yml|yaml)$/",
```

Comment:
We might be able to hook this up directly to the template registry using `setContext`. For example, every time the registry updates we set a custom context (within the registry class):
```
vscode.commands.executeCommand('setContext', 'aws.samTemplates', [...this.registryData.keys()])
```
Then the `when` clause can check it:
```
"when": "isFileSystemResource == true && resourceFilename in aws.samTemplates",
```

Reply:
I'd love to explore this further since I'm thinking https://github.com/aws/aws-toolkit-vscode/pull/1748 will be the first significant steps towards us supporting more than just `template.yaml` files.

Reply:
Good idea. We can explore this later, don't want to block this initial effort though.